symbolic-common/src/path.rs

```diff
@@ -92,54 +92,95 @@ fn is_windows_path<P: AsRef<[u8]>>(path: P) -> bool {
 /// Join a relative UNIX path:
 ///
 /// ```
-/// assert_eq!(symbolic_common::join_path("/a/b", "c/d"), "/a/b/c/d");
+/// let mut base = String::from("/a/b");
+/// symbolic_common::join_path_buf(&mut base, "c/d");
+/// assert_eq!(&base, "/a/b/c/d");
 /// ```
 ///
 /// Join a Windows drive letter path path:
 ///
 /// ```
-/// assert_eq!(symbolic_common::join_path("C:\\a", "b\\c"), "C:\\a\\b\\c");
+/// let mut base = String::from("C:\\a");
+/// symbolic_common::join_path_buf(&mut base, "b\\c");
+/// assert_eq!(&base, "C:\\a\\b\\c");
 /// ```
 ///
 /// If the right-hand side is an absolute path, it replaces the left-hand side:
 ///
 /// ```
-/// assert_eq!(symbolic_common::join_path("/a/b", "/c/d"), "/c/d");
+/// let mut base = String::from("/a/b");
+/// symbolic_common::join_path_buf(&mut base, "/c/d");
+/// assert_eq!(&base, "/c/d");
 /// ```
-pub fn join_path(base: &str, other: &str) -> String {
+pub fn join_path_buf(base: &mut String, other: &str) {
```

Comment:
an alternative would be `fn join_path_buf(base: String, other: &str) -> String`, but I think that does not matter that much.